### PR TITLE
Initial version

### PR DESCRIPTION
**Pull request by PyDataBlog**
- [x] Implement DictDB template
- [x] Add tests
- [x] Implement searching algorithm
- [ ] Add docs

### CODE REVIEW

File: src/SimString.jl [hunk 1]
```diff
@@ -1,5 +1,26 @@
...
+include("measures.jl")
+include("search.jl")
+
+
+
+####### Global export of user API #######
+export Dice, Jaccard, Cosine,
+    AbstractSimStringDB, DictDB,
+    CharacterNGrams, WordNGrams,
+    push!, append!, search
```
Comment (Arkoniak) [inline src/SimString.jl]: You shouldn't export `Base` function.

File: src/db_collection.jl [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+"""
+abstract type FeatureExtractor end
+
+
+"""
+Custom DB collection for storing SimString data using base Dictionary `Dict`
+"""
+struct DictDB{
+    T1<:FeatureExtractor,
+    T2<:AbstractString,
```
Comment (Arkoniak) [inline src/db_collection.jl:20]: You shouldn't use abstract types in struct definition

File: src/SimString.jl (hunk 1 above)
Comment (PyDataBlog) [reply src/SimString.jl]: I'll update them then

File: src/features.jl [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+
+"""
+Internal function to count and pad generated character-level ngrams (including duplicates)
+"""
+function cummulative_ngram_count(extractor::WordNGrams, x)
+    p1 = sortperm(x)
+    p2 = sortperm(p1)
+    x = sort(x)
+
+    results = []
```
Comment (Arkoniak) [inline src/features.jl]: type unstable definition of `results`. You should define properly typed array.

File: src/features.jl [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+    counter = 0
+    last_i, rest = Iterators.peel(x)
+
+    push!(last_i, "#$(counter += 1)")
+    push!(results, last_i)
+
+    for i in rest
+        counter = i == last_i[1:extractor.n] ? counter + 1 : 1
```
Comment (Arkoniak) [inline src/features.jl]: It makes no sense.
`last_i` is scalar, `last_i[1:extractor.n]` should error.

File: src/features.jl [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+    counter = Dict{String, Int}()
+    unique_list = String[]
+
+    for val in x
+        if val in keys(counter)
+            counter[val] += 1
+        else
+            counter[val] = 1
+        end
+        push!(unique_list, string(val, "#", counter[val]))
```
Comment (Arkoniak) [inline src/features.jl]: `push!` is probably slowing this thing down. It's better to preallocate the result and assign by index.

File: src/features.jl (hunk 4 above)
Comment (PyDataBlog) [reply src/features.jl]: Actually this method is for dispatching on word grams so it won’t fail as `last_i` is a vector and not a scalar.

**Pull request merged**